spec/controllers/http_status_agent_spec.rb

```diff
@@ -190,8 +199,8 @@ def agent.checked_url
       describe "but the ping returns a status code of -1" do
 
         let(:event_with_a_successful_ping) do
-          agent.faraday.set(successful_url, Struct.new(:status).new(-1))
-          Event.new.tap { |e| e.payload = { url: successful_url } }
+          agent.faraday.set(successful_url, Struct.new(:status, :headers).new(-1, {}))
+          Event.new.tap { |e| e.payload = { url: successful_url, headers_to_save: "" } }
```

Comment:
The `headers_to_save` of `agent` is set to `"Server"`, so feeding this value via Event does not affect the option value.  You need either set the option to `{% if headers_to_save %}{{ headers_to_save }}{% else %}Server{% endif %}` (`""` is a true value in Liquid) or fix the Agent implementation to prefer a value in Event over one in options.

The specs are wrongly passing because `agent.interpolated` is mocked to return what's in the Event ignoring the options, which is not the case in reality.